packages/flutter/lib/src/native/cocoa/sentry_native_cocoa.dart

```diff
@@ -122,226 +73,11 @@ class SentryNativeCocoa extends SentryNativeChannel {
   }
 
   @override
-  int? startProfiler(SentryId traceId) => tryCatchSync(
-        'startProfiler',
-        () {
-          final sentryId$1 = cocoa.SentryId.alloc()
-              .initWithUUIDString(NSString(traceId.toString()));
-
-          final sentryId = cocoa.SentryId.castFromPointer(
-            sentryId$1.ref.pointer,
-            retain: true,
-            release: true,
-          );
-
-          final startTime =
-              cocoa.PrivateSentrySDKOnly.startProfilerForTrace(sentryId);
-          return startTime;
-        },
-      );
-
-  @override
-  int? displayRefreshRate() => tryCatchSync(
-        'displayRefreshRate',
-        () {
-          final refreshRate = cocoa.SentryFlutterPlugin.getDisplayRefreshRate();
-          return refreshRate?.intValue;
-        },
-      );
-
-  @override
-  NativeAppStart? fetchNativeAppStart() => tryCatchSync(
-        'fetchNativeAppStart',
-        () {
-          final appStartUtf8JsonBytes =
-              cocoa.SentryFlutterPlugin.fetchNativeAppStartAsBytes();
-          if (appStartUtf8JsonBytes == null) return null;
-
-          final json = decodeUtf8JsonMap(appStartUtf8JsonBytes.toList());
-          return NativeAppStart.fromJson(json);
-        },
-      );
-
-  @override
-  void addBreadcrumb(Breadcrumb breadcrumb) =>
-      tryCatchSync('addBreadcrumb', () {
-        final nativeBreadcrumb =
-            cocoa.PrivateSentrySDKOnly.breadcrumbWithDictionary(
-                _dartToNSDictionary(breadcrumb.toJson()));
-        cocoa.SentrySDK.addBreadcrumb(nativeBreadcrumb);
-      });
-
-  @override
-  void clearBreadcrumbs() => tryCatchSync('clearBreadcrumbs', () {
-        cocoa.SentrySDK.configureScope(
-            cocoa.ObjCBlock_ffiVoid_SentryScope.fromFunction(
-                (cocoa.SentryScope scope) {
-          scope.clearBreadcrumbs();
-        }));
-      });
-
-  @override
-  void nativeCrash() => cocoa.SentrySDK.crash();
-
-  @override
-  void pauseAppHangTracking() => tryCatchSync('pauseAppHangTracking', () {
-        cocoa.SentrySDK.pauseAppHangTracking();
-      });
-
-  @override
-  void resumeAppHangTracking() => tryCatchSync('resumeAppHangTracking', () {
-        cocoa.SentrySDK.resumeAppHangTracking();
+  int? startProfiler(SentryId traceId) => tryCatchSync('startProfiler', () {
+        final cSentryId = cocoa.SentryId1.alloc(_lib)
+          ..initWithUUIDString_(cocoa.NSString(_lib, traceId.toString()));
+        final startTime =
+            cocoa.PrivateSentrySDKOnly.startProfilerForTrace_(_lib, cSentryId);
```

Comment:
Are we still leaving some parts handled through FFI? Wouldn't we have the same issue then?